rectools/visuals/metrics_app.py

```diff
@@ -85,125 +97,274 @@ def construct(
+    def fold_ids(self) -> tp.Optional[tp.List[int]]:
         """Sorted list of fold identifiers from the `models_metrics`."""
-        return sorted(self.models_metrics[Columns.Split].unique())
-
-    def _validate_input_data(self) -> None:
-        if not isinstance(self.models_metrics, pd.DataFrame):
-            raise ValueError("Incorrect input type. `metrics_data` should be a DataFrame")
-        if Columns.Split not in self.models_metrics.columns:
-            raise KeyError("Missing `Split` column in `metrics_data` DataFrame")
-        if Columns.Model not in self.models_metrics.columns:
-            raise KeyError("Missing `Model`` column in `metrics_data` DataFrame")
-        if len(self.models_metrics.columns) < 3:
+        if Columns.Split in self.data.columns:
+            return sorted(self.data[Columns.Split].unique())
+        return None
+
+    @staticmethod
+    def _validate_models_metrics_base(models_metrics: pd.DataFrame) -> None:
+        if Columns.Model not in models_metrics.columns:
+            raise KeyError("Missing `Model` column in `metrics_data` DataFrame")
+        if len([item for item in models_metrics.columns if item not in {Columns.Model, Columns.Split}]) < 1:
             raise KeyError("`metrics_data` DataFrame assumed to have at least one metric column")
+        if models_metrics[Columns.Model].isnull().values.any():
+            raise ValueError("Found NaN values in `Model`column of `metrics_data`")
+        if Columns.Split in models_metrics.columns and models_metrics[Columns.Split].isnull().values.any():
+            raise ValueError("Found NaN values in `Split`column of `metrics_data`")
+
+    @staticmethod
+    def _validate_models_metrics_split(models_metrics: pd.DataFrame) -> None:
+        # Validate that each model have same folds names
+        if Columns.Split not in models_metrics.columns:
+            return
+        grouped = models_metrics.groupby(Columns.Model)
+        # get the first group's fold counts and fold names for reference
+        ref_group_name = next(iter(grouped.groups.keys()))
+        ref_group = grouped.get_group(ref_group_name)
+        ref_fold_count = ref_group[Columns.Split].nunique()
+        ref_fold_names = set(ref_group[Columns.Split].unique())
+
+        for model, group in grouped:
+            if group["i_split"].nunique() != ref_fold_count:
+                raise ValueError(f"""{model} does not have the same fold amount as {ref_group_name}""")
+            if set(group["i_split"].unique()) != ref_fold_names:
+                raise ValueError(f"{model} does not have the same fold names as {ref_group_name}")
+
+    @staticmethod
+    def _validate_models_metrics_names(models_metrics: pd.DataFrame) -> None:
+        # Validate that all Models names are unique
+        models_metrics = models_metrics.copy()
+        if Columns.Split in models_metrics.columns:
+            models_names_comb = models_metrics[Columns.Model] + models_metrics[Columns.Split].astype(str)
```

Comment:
Really dangerous approach. Imagine you have next rows:  

Model | Split
------+------
a         | bc
------+-----
ab      | c

This approach won't work correctly. 
Better use `duplicated` or `groupby`

Reply:
Fixed